### ISSUE TITLE
You could probably add a check for positive INT format of .index field

### ISSUE THREAD

**Issue by DanNBullock** [brainlife/validator-neuro-wmc#8]
Because the .index field is explicitly associated with the .names field, there can't be any non integer values.  Also, because the length of the .names field is necessarily 0 or more, none of the .index values can be negative.

**Comment by soichih** [comment]
A bad app can create .index array with data like this

>  0     0     0     0     0    26     0    15     0     0     **-1**     0     0     0     0     0     0     0

I don't think this ever happened, but it might be good to make sure .names array contains an entry at the value (base-1) in this array points to. That will in effect make sure negative index doesn't exist.

**Comment by DanNBullock** [comment]
> A bad app can create .index array with data like this
> 
> > 0     0     0     0     0    26     0    15     0     0     **-1**     0     0     0     0     0     0     0
> 
> I don't think this ever happened, but it might be good to make sure .names array contains an entry at the value (base-1) in this array points to. That will in effect make sure negative index doesn't exist.

I don't think we should allow an app to do that.  I'm not sure how we could enforce that, but it would be ideal.

In essence, what ought to be able to happen is that an algorithm (i.e. https://github.com/DanNBullock/wma_tools/blob/master/ClassificationStruc_Tools/bsc_makeFGsFromClassification_v5.m) should be able to iterate across the unique entries of .index and extract tract for each unique value (i.e. https://github.com/DanNBullock/wma_tools/blob/f989c66dbfcc0bb943e9f1ed0954a6d3b52ab32a/ClassificationStruc_Tools/bsc_makeFGsFromClassification_v5.m#L62)

In the case of a negative number, this sort of algorithm is going to try and index into the -Nth entry of the name index and, if I understand matlab indexing, this will cause an error (or will it simply wrap around?).

I'm not sure how to interpret this suggestion:
"might be good to make sure .names array contains an entry at the value (base-1)".  
I don't know what it means to store a value at a negative location in a matlab array.

**Comment by soichih** [comment]
> I don't think we should allow an app to do that. I'm not sure how we
could enforce that, but it would be ideal.
We can't enforce Apps to not do it, but we can add a check on our
validator.

> I'm not sure how to interpret this suggestion: "might be good to make
sure .names array contains an entry at the value (base-1)". I don't know
what it means to store a value at a negative location in a matlab array.
For example, we can go through each value in .index, and try to find the
item in .names. If it fails with KeyError, that means that item doesn't
exist. But I think simply checking the min/max value should be enough. ie..
`np.max(index) <= len(names)`

**Comment by DanNBullock** [comment]
I think that would work given that the min len of (.names) is 0, and the max correspondence is pretty straightforward.  

Not sure that fixes the int vs float issue though.  Though, I don't know what app would output a .5 in the index file either.
